pkg/netdevice/netDeviceProvider.go

```diff
@@ -209,6 +209,21 @@ func (np *netDeviceProvider) GetFilteredDevices(devices []types.PciDevice, rc *t
 		filteredDevice = rdmaDevices
 	}
 
+	// filter for vDPA-capable devices
+	if nf.VdpaType != "" {
```

Comment:
I noticed your comment that isRdma and vdpaType are mutually exclusive, shall this line be:
```
else if nf.VdpaType != "" {
```

Reply:
@amorenoz wdyt?

Reply:
I think if we add a basic check for the resource config in `initServers()` (add a validateResourceConfig() check) we can keep it as is here. this bit of code should not be aware of these dependencies

Reply:
OK, I'll add a check in `resourceManager::validConfigs`